src/helics/application_api/Federate.hpp

```diff
@@ -483,6 +485,27 @@ class Federate
     @return a const ref to  std::string containing the units */
     const std::string &getInterfaceUnits (interface_handle handle) const;
 
+    /** log a message to the federate Logger
+   @param level the logging level of the message
+   @param logMessageSource- the name of the object that sent the message
+   @param message the message to log
+   */
+    virtual void logMessage (int level, const std::string &logMessageSource, const std::string &message) const;
+
+    /** log a simplified message to the federate Logger
+    @param message the message to log
+    */
+    virtual void logMessage (const std::string &message) const;
+
+    /** set the logging function
+    @details function must have signature void(int level, const std::string &sourceName, const std::string
+    &message)
+    */
+    virtual void setLogger (std::function<void(int, const std::string &, const std::string &)> logFunction)
```

Comment:
this would seem a duplicate function to setLoggingCallback,  Are you thinking there is some distinction between this logger and the callback used in setLoggingCallback which goes to the core?